src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User userFromDB = userService.findByEmail(email).orElseThrow(() -> 
+                new AuthenticationException("Can't find user by this email: " + email));
+        String userFromDbPassword = userFromDB.getPassword();
+        String currentUserHashedPassword = HashUtil.hashPassword(password, userFromDB.getSalt());
+        if (currentUserHashedPassword.equals(userFromDbPassword)) {
+            return userFromDB;
+        } 
+        throw new AuthenticationException("Your password not match. Try again.");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Unable to register a new user with this "
+                    + "email address: " + email
+                    + ". It has already been used for registration");
+        }
+
+        if (password.length() <= 6) {
+            throw new RegistrationException("The password must be 6 or more characters.");
+        }
```

Comment:
6 should be constant
but better delete this at all - we will have task for data validation